cloudnet-modules/cloudnet-database-mongodb/src/main/java/de/dytanic/cloudnet/ext/database/mongodb/MongoDatabaseProvider.java

```diff
@@ -0,0 +1,149 @@
+package de.dytanic.cloudnet.ext.database.mongodb;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import de.dytanic.cloudnet.common.concurrent.ITask;
+import de.dytanic.cloudnet.common.concurrent.ListenableTask;
+import de.dytanic.cloudnet.common.document.gson.JsonDocument;
+import de.dytanic.cloudnet.database.AbstractDatabaseProvider;
+import de.dytanic.cloudnet.driver.CloudNetDriver;
+import de.dytanic.cloudnet.driver.database.Database;
+import de.dytanic.cloudnet.ext.database.mongodb.util.DatabaseConnectionData;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+@Getter
+public class MongoDatabaseProvider extends AbstractDatabaseProvider {
+
+    private MongoClient client;
+    private JsonDocument config;
+    private DatabaseConnectionData databaseConnectionData;
+    private ExecutorService executorService;
+
+    public MongoDatabaseProvider(JsonDocument config, ExecutorService executorService) {
+        this.config = config;
+        this.executorService = executorService;
+        this.databaseConnectionData = DatabaseConnectionData.fromJsonDocument(config);
+    }
+
+    public void closeSession() {
+        try {
+            getClient().close();
+        } catch (Exception e) {
+        }
+    }
+
+    public boolean isConnected() {
+        return getClient() != null;
+    }
+
+    @Override
+    public boolean init() throws Exception {
+        Executors.newScheduledThreadPool(1).scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                if (!isConnected()) {
```

Comment:
This checks if the client is null, but close session doesn't set the client to null...